corva/app/context.py

```diff
@@ -0,0 +1,19 @@
+from pydantic import BaseModel
```

Comment:
can we add a docstring. What is it `context` in the scope of this project and why do we need it?

Reply:
Q1: What is the purpose of the python-sdk?
A1: Help companies develop custom apps easier.

Q2: Why can't Corva develop custom apps for companies?
A2: Companies have secret proprietary algorithms that they don't want to expose. The best Corva can do is to provide a tool for companies to easily write their custom apps.

Q3: What types of custom apps do companies want?
A3: Stream, Scheduled and Task apps.

Q4: What is BaseApp class in python-sdk?
A4: Each type of app executes three main steps: pre_process, process and post_process. BaseApp provides an basic interface, that each subclass (e.g TaskApp) is able to extend.

Q5: What was the problem that Context class solved?
A5: Each child class of BaseApp needed unique set of parameters to run (e.g. StreamApp needed state param, whereas TaskApp needed task_data param). Also TaskApp needed a way to pass additional data from `process` to `post_process`.

Problem:
```python
class BaseApp:
    def run(self):
        event = self.get_event()
        state = self.get_state()  # how to initialize for TaskApp that doesn't need this
        task_data = self.get_task_data()  # how to initialize for StreamApp that doesn't need this?

        self.pre_process(event=event, state=state, task_data=task_data)
        self.process(event=event, state=state, task_data=task_data)
        self.post_process(event=event, state=state, task_data=task_data)
    
    # bloated function calls below
    def pre_process(self, event, state, task_data): ...
    def process(self, event, state, task_data): ...
    def post_process(self, event, state, task_data): ...

class StreamApp(BaseApp):
    # no need in task_data param in below functions
    def pre_process(self, event, state, **kwargs): ...
    def process(self, event, state, **kwargs): ...
    def post_process(self, event, state, **kwargs): ...

class TaskApp(BaseApp):
    # no need in state param in below functions
    def pre_process(self, event, task_data, **kwargs): ...
    def process(self, event, task_data, **kwargs): ...
    def post_process(self, event, task_data, **kwargs): ...
```
Solution:
```python
class BaseContext:
    event: 'Event'

class StreamContext(BaseContext):
    state: 'State'

class TaskContext(BaseContext):
    task_data: 'TaskData'
    process_result: dict = {}  # will be set in process function

class BaseApp:
    def run(self):
        context = self.get_context()
        
        # clean function calls below
        self.pre_process(context)
        self.process(context)
        self.post_process(context)

    def get_context(self) -> BaseContext:
        """Returns all params needed for app to run in a BaseContext object"""
        ...

    def pre_process(self, context: BaseContext): ...
    def process(self, context: BaseContext): ...
    def post_process(self, context: BaseContext): ...

class StreamApp(BaseApp):
    def get_context(self) -> BaseContext: ... # returns StreamContext, knows how to initialize it

    # below functions only get parameters that they actually need
    def pre_process(self, context: BaseContext): ...
    def process(self, context: BaseContext): ...
    def post_process(self, context: BaseContext): ...

class TaskApp(BaseApp):
    def get_context(self) -> BaseContext: ...  # returns TaskContext, knows how to initialize it

    # below functions only get parameters that they actually need
    def pre_process(self, context: BaseContext): ...
    def process(self, context: BaseContext): ...
    def post_process(self, context: BaseContext): ...
```
In the same way companies will be able to write their own apps, with custom parameters passed to process function using Context.

Reply:
Also added a docstring to BaseContext

Reply:
perfect! Thank you!

The one comment: let's keep context near the class itself. We don't really need another module to keep all contexts

Reply:
I've moved these models to appropriate corva/models/ dirs in [pull 7](https://github.com/corva-ai/python-sdk/pull/7)